src/main/java/org/mapstruct/intellij/codeinsight/references/MapstructMappingQualifiedByNameReference.java

```diff
@@ -0,0 +1,173 @@
+            .filter( this::methodHasReturnType )
+            .filter( m -> findAnnotation( m, true, MapstructUtil.NAMED_ANNOTATION_FQN ) != null )
+            .filter( a -> Objects.equals( getNamedValue( a ), value ) )
+            .findAny()
+            .orElse( null );
+    }
+
+    @Nullable
+    private String getNamedValue(PsiMethod method) {
+
+        PsiAnnotation annotation = findAnnotation( method, true, MapstructUtil.NAMED_ANNOTATION_FQN );
+
+        if ( annotation == null ) {
+            return null;
+        }
+
+        return getStringAttributeValue( annotation, "value" );
+    }
+
+    @NotNull
+    @Override
+    Object[] getVariantsInternal(@NotNull PsiType psiType) {
+        return LookupElement.EMPTY_ARRAY; // not needed
+    }
+
+    @NotNull
+    @Override
+    Object[] getVariantsInternal(@NotNull PsiMethod mappingMethod) {
+
+        List<PsiMethod> allMethodsFromThisAndReferencedMappers = findAllMethodsFromThisAndReferencedMappers(
+            mappingMethod );
+
+        return allMethodsFromThisAndReferencedMappers.stream()
+            .filter( MapstructUtil::isNamedMethod )
+            .filter( this::methodHasReturnType )
+            .map( this::methodAsLookup )
+            .toArray();
+    }
+
+    private boolean methodHasReturnType(@NotNull PsiMethod psiMethod) {
+        return !PsiType.VOID.equals( psiMethod.getReturnType() );
+    }
+
+    @NotNull
+    private List<PsiMethod> findAllMethodsFromThisAndReferencedMappers(@NotNull PsiMethod mappingMethod) {
+
+        PsiClass containingClass = mappingMethod.getContainingClass();
+        if ( containingClass == null ) {
+            return Collections.emptyList();
+        }
+
+        Stream<PsiMethod> internalMethods = Stream.of( containingClass.getMethods() );
+
+        Stream<PsiMethod> externalMethods = findNamedMethodsInUsedMappers( containingClass );
+
+        return Stream.concat( internalMethods, externalMethods ).collect( Collectors.toList() );
+    }
+
+    @NotNull
+    private Stream<PsiMethod> findNamedMethodsInUsedMappers(@Nullable PsiClass containingClass) {
+
+        PsiAnnotation mapperAnnotation = findAnnotation(
+            containingClass,
+            MapstructUtil.MAPPER_ANNOTATION_FQN
+        );
+
+        if ( mapperAnnotation == null ) {
+            return Stream.empty();
+        }
+
+        List<PsiClass> externalMappers = MapstructAnnotationUtils.resolveUsesConfigClasses( mapperAnnotation );
+        if ( externalMappers == null ) {
+            return Stream.empty();
+        }
+
+        return externalMappers.stream()
+            .flatMap( psiClass -> Arrays.stream( psiClass.getMethods() ) );
+    }
+
+    private LookupElement methodAsLookup(@NotNull PsiMethod method) {
+        String lookupString = getNamedValue( method );
+        return asLookupWithRepresentableText(
+            method,
+            lookupString,
+            lookupString,
+            String.format(
```

Comment:
I think that for this it would be good to show the method parameters here as well. e.g. `StringMapper#trim(String)`.